src/mlpack/methods/reinforcement_learning/q_networks/dueling_dqn.hpp

```diff
@@ -0,0 +1,124 @@
+/**
+ * @file methods/reinforcement_learning/q_networks/dueling_dqn.hpp
+ * @author Nishant Kumar
+ *
+ * This file contains the implementation of the dueling deep q network.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_RL_DUELING_DQN_HPP
+#define MLPACK_METHODS_RL_DUELING_DQN_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/methods/ann/ffn.hpp>
+#include <mlpack/methods/ann/init_rules/gaussian_init.hpp>
+#include <mlpack/methods/ann/layer/layer.hpp>
+#include <mlpack/methods/ann/loss_functions/mean_squared_error.hpp>
+
+namespace mlpack {
+namespace rl {
+
+using namespace mlpack::ann;
+
+/**
+ * @tparam NetworkType The type of network used for dueling dqn.
+ */
+template <typename NetworkType = FFN<EmptyLoss<>,
```

Comment:
Is there a better way to create a FFN with no loss at the end?

Reply:
No, but the compiler should optimize the empty function out.

Reply:
should I add this : https://github.com/mlpack/mlpack/pull/1912/commits/b5b72a0d81840b7d0d5e9584aa3ce47622703517

Reply:
Yes, good idea.